packages/flutter/lib/src/widgets/routes.dart

```diff
@@ -143,8 +133,10 @@ abstract class TransitionRoute<T> extends OverlayRoute<T> {
         break;
       case AnimationStatus.dismissed:
         assert(!overlayEntries.first.opaque);
-        finished(); // clear the overlays
-        assert(overlayEntries.isEmpty);
+        if (!isCurrent) {
```

Comment:
iOS back gesture.  I can add a comment to that effect.

When you slide back, the animations goes all the way to "dismissed" and then the route gets popped.